### DISCUSSION TITLE
Convert a saved editorState to plain text without editor?

### DISCUSSION THREAD

**Discussion by francistogram** [facebook/lexical#3806] [Q&A]
I found [this discussion](https://github.com/facebook/lexical/discussions/1934) which suggests to get the text content from the editor is using the rootNode's `getTextContent()`

```javascript
import {
  $getRoot,
} from 'lexical';

// ... 

const stringifiedEditorState = JSON.stringify(
  editor.getEditorState().toJSON(),
);
const parsedEditorState = editor.parseEditorState(stringifiedEditorState);

const editorStateTextString = parsedEditorState.read(() => $getRoot().getTextContent())
```

I'm wondering if it's possible to do this without the editor instance. The use case I have is scheduling tweets and you'd draft it from the editor and then saved to the backend as JSON but when it's time to post I'd need the text content

While it'd work to save both the editor state and the text content I'm hesitant to take that path since they should always be in sync and was hoping some utility existed to do something like `convertEditorStateToTextContent(editorState: EditorState): string` but I could be convinced to save the string too I suppose

**Comment by moy2010** [comment]
If you know the schema, you can write your own logic to convert the editor state to plain text.

**Comment by kevinjames95** [comment]
editor.getEditorState().read(() => $getRoot().getTextContent()) will also work

**Comment by kunKun-tx** [comment]
I found a solution to convert state to plain text **without editor**. My use case is to performing on-page indexing using lexical editor state saved from Payload CMS. The plaintext works much better than other options.

Two steps:
1. You first convert the saved editor state to html using [`convertLexicalToHTML`](https://lexical.dev/docs/packages/lexical-html)
2. And then use `innerHTML` to get the text. 

```js
const html = DOMPurify.sanitize(
convertLexicalToHTML({ data: x.description }),
)
const div = document.createElement("div")
div.innerHTML = html
const text = div.textContent || ""
```

Notes: 
- DOMPurify probably is overkill here, but can't be too cautious.
- with the help of `jsdom`, you can probably do this with nodejs as well.
